fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -873,6 +872,51 @@ public Void visitFunctionCall(FunctionCallExpr node, Scope scope) {
                     throw new SemanticException(fnName + " should have at least two inputs");
                 }
                 fn = Expr.getBuiltinFunction(fnName, argumentTypes, Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+            } else if (fnName.equals("array_generate")) {
+                if (node.getChildren().size() < 1 || node.getChildren().size() > 3) {
+                    throw new SemanticException(fnName + " has wrong input numbers");
+                }
+                for (Expr expr : node.getChildren()) {
+                    if ((expr instanceof SlotRef) && node.getChildren().size() != 3) {
+                        throw new SemanticException(fnName + " with IntColumn doesn't support default parameters");
+                    }
+                    if (!(expr instanceof IntLiteral) && !(expr instanceof LargeIntLiteral) &&
+                            !(expr instanceof SlotRef) && !(expr instanceof NullLiteral)) {
+                        throw new SemanticException(fnName + "'s parameter only support Integer");
+                    }
+                }
+                // add the default parameters for array_generate
+                if (node.getChildren().size() == 1) {
+                    LiteralExpr secondParam = (LiteralExpr) node.getChild(0);
+                    node.clearChildren();
+                    try {
+                        node.addChild(new IntLiteral("1", Type.TINYINT));
+                        node.addChild(secondParam);
+                    } catch (AnalysisException e) {
+                        throw new SemanticException(e.getMessage());
+                    }
+                }
+                if (node.getChildren().size() == 2) {
+                    int idx = 0;
+                    BigInteger[] childValues = new BigInteger[2];
+                    for (Expr expr : node.getChildren()) {
+                        if (expr instanceof NullLiteral) {
```

Comment:
exprs as follows are all legal, throw exception is not acceptable
array_generate(NULL)
array_generate(NULL, 10)
array_generate(1, NULL)
array_generate(1， 10， NULL)

Reply:
I will fix it